### ISSUE TITLE
Add scaled_dot_product_attention "scale" argument to nn.MultiHeadAttention 

### ISSUE THREAD

**Issue by F-Barto** [pytorch/pytorch#124718]
### 🚀 The feature, motivation and pitch

Recently the Maximal Update Parametrization ([muP, arxiv 2203.03466](https://arxiv.org/abs/2203.03466)) is becoming prevalent in large model training becaus of its "zero-shot hyperparameter" transfer capabilities (i.e., grid search the best HPs on 2-10M params models and and train the 2B+ params with the same HPs)

Refs:
- [FLM-101B: An Open LLM and How to Train It with $100K Budget](https://arxiv.org/abs/2309.03852)
- [Cerebras-GPT: Open Compute-Optimal Language Models Trained on the Cerebras Wafer-Scale Cluster](https://arxiv.org/abs/2304.03208)
- [JetMoE: Reaching Llama2 Performance with 0.1M Dollars](https://arxiv.org/abs/2404.07413)

One modification involved in getting a transformer muParametrized is to modify the softmax scaling from $1/\sqrt{d}$ to $1/d$ or $1/d*\alpha$ with $\alpha$ being grid searched/learnable

This can easily be done with `torch.nn.functional.scaled_dot_product_attention` using its "[scale](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html#torch.nn.MultiheadAttention)" argument, but nothing like that exists for [torch.nn.MultiheadAttention](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html#torch.nn.MultiheadAttention).

**Comment by clorenz7** [comment]
I would be interested in working on this.

**Comment by drisspg** [comment]
Is there a reason why you wouldnt just using sdpa directly instead of using nn.MHA
